R/seroprevalence_data.R

```diff
@@ -120,4 +118,184 @@ prepare_bin_data <- function(serodata) {
+                              ){
+    sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
+        mutate(tsur = tsur,
+            country = 'None',
+            test = test,
+            antibody = antibody,
+            survey = survey_label,
+            age_mean_f = tsur - birth_year)
+    sim_n_seropositive <- get_sim_n_seropositive(sim_data, foi, sample_size_by_age, seed = seed)
+    sim_data <- sim_data %>%
+        mutate(counts = sim_n_seropositive$n_seropositive,
+               total = sample_size_by_age) %>%
+        prepare_serodata(add_age_mean_f = FALSE)
+
+    return(sim_data)
+}
+
+#' Method for constructing age-group variable from age column
+#'
+#' This function was taken from \link[vaccineff]{get_age_group}.
+#' This method splits an age interval from age_min to age_max into
+#' (age_max-age_min)/step intervals.
+#' By default age_min is set 0, however it can be assigned by
+#' convenience.
+#' If the method finds ages greater or equal than age_max
+#' it assigns the string ">{age_max}".
+#' To avoid errors it is necessary to set step < age_max.
+#' It is also suggested to choose the step such
+#' that age_max%%(step+1) = 0.
+#' @param age vector containing age information
+#' @param  step step used to split the age interval
+#' @return age_group factor variable grouping \code{age} by the age intervals 
+#' specified by \code{min(age)}, \code{max(age)}.
+get_age_group <- function(age, step) {
+  age_min <- min(age)
+  age_max <- max(age)
+  n_steps <- as.integer((age_max - age_min) / step) + 1
+  limits_low <- c(as.integer(seq(age_min,
+                                 age_max,
+                                 length.out = n_steps)))
+  limits_hgh <- limits_low + step
+  lim_labels <- paste(as.character(limits_low), as.character(limits_hgh),
+                      sep = "-")
+  lim_labels[length(lim_labels)] <- paste0("+",
+                                           limits_low[length(limits_low)])
+  lim_breaks <- c(-Inf, limits_low[2:length(limits_low)] - 1, Inf)
+
+  age_group <- cut(age,
+                   breaks = lim_breaks,
+                   labels = lim_labels,
+                   # this is for the intervals to be closed to the left and open to the right
+                   right = FALSE) 
+  return(age_group)
+}
+
+#' Function that groups a simulated serological dataset by age
+#'
+#' @param sim_data Dataframe with the same structure as the output of \code{\linl{generate_sim_data}}.
+#' @param col_age name of the column containing the age information
+#' @param step step used to split the age interval
+#' @return Dataframe object containing grouped simulated data generated from \code{foi}
+group_sim_data <- function(sim_data,
+                          col_age = "age_mean_f",
+                          step = 5) {
+    age <- sim_data[[col_age]]
+    sim_data$age_group <-  get_age_group(age = age, step = step)
+    sim_data_grouped <- sim_data %>% group_by(age_group) %>%
```

Comment:
Minor thing but can we move the `group_by` to the next line for consistency when using the pipe?